### PR TITLE
feat: use react

### PR DESCRIPTION
**Pull request by cbt1**
Removes the react-native implementation and replaces with a react implementation. No more is support for native platforms.

### CODE REVIEW

File: src/pivot-table/components/CellFactory.tsx [hunk 1]
```diff
@@ -1,87 +1,52 @@
 import React from 'react';
-import { View, Text, StyleSheet } from 'react-native';
-import { Model } from '../../types/types';
-import { Cell, TYPE } from '../handle-data';
+import { ItemData } from '../../types/types';
+import { TYPE } from '../handle-data';
```
Comment (niekvanstaveren) [inline src/pivot-table/components/CellFactory.tsx:3]: TYPE looks like an enum. Not something that needs to be changed in this PR though. Also there might be a better name for it 😄

File: src/pivot-table/components/DimensionCell.tsx [hunk 2]
```diff
@@ -1,26 +1,30 @@
...
+import { borderStyle, textStyle } from './shared-styles';
 
 export interface DimensionCellProps {
   cell: Cell;
   model: Model;
   rowIndex: number;
   colIndex: number;
   isLeftColumn?: boolean;
-  style: ViewStyle | ViewStyle[];
+  style: any;
```
Comment (niekvanstaveren) [inline src/pivot-table/components/DimensionCell.tsx:13]: We should type this as well. Does this contain all the styling for a Cell?

File: src/pivot-table/components/DimensionCell.tsx [hunk 3]
```diff
@@ -39,19 +43,15 @@ const DimensionCell = ({ model, cell, isLeftColumn = false, rowIndex = 0, colInd
...
-  const DimCell = (
-    <View style={style}>
-      <Text
-        style={[sharedStyles.text, styles.text]}
-        numberOfLines={1}>
-          {cellContent}
-      </Text>
-    </View>
+  return (
+    <div style={{ ...style, ...containerStyle}}>
```
Comment (niekvanstaveren) [inline src/pivot-table/components/DimensionCell.tsx:47]: Would it make any sense creating a real table now that we don't do mobile native anymore?

File: src/types/stardust.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module '@nebula.js/stardust';
```
Comment (niekvanstaveren) [inline src/types/stardust.d.ts:1]: We should check with Thomas on the state of stardust types being exported.

**Review by niekvanstaveren — APPROVED** [review]
Looks good from a quick view. Some notes that are not directly related to this PR, so can also be handled later

File: src/types/stardust.d.ts (hunk 4 above)
Comment (cbt1) [reply src/types/stardust.d.ts:1]: Agree. I just made dirty quick fix to remove the type error.

File: src/pivot-table/components/DimensionCell.tsx (hunk 3 above)
Comment (cbt1) [reply src/pivot-table/components/DimensionCell.tsx:47]: Not while we are using the `react-window` component. There each cell get an absolute position. Which allows us to only render part of a larger dataset.

File: src/pivot-table/components/DimensionCell.tsx (hunk 2 above)
Comment (cbt1) [reply src/pivot-table/components/DimensionCell.tsx:13]: It should be typed yes. It only contains the absolute position of the cell. Which is coming from the `react-window` component.

File: src/pivot-table/components/CellFactory.tsx (hunk 1 above)
Comment (cbt1) [reply src/pivot-table/components/CellFactory.tsx:3]: Yeah it should be an enum. Not sure why I did do it like an object. Might have be that I ran into to some issue. Will take another look at it.

**Pull request merged**